### PR TITLE
Create vscode_substrate_extension.md

### PR DESCRIPTION
**Pull request by bytemaster333**
#### Project Abstract

> Please replace these instructions with a brief description of your project summarising key points (1-2 paragraphs).
>
> If your application is a follow-up to a previous grant, please mention which one in the first line of the abstract and include a link to previous pull requests if applicable.

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [X] **Level 2**:  Up to $30,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [X] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/grant_guidelines_per_category.md).
- [X] Payment details have been provided (Polkadot AssetHub (USDC & DOT) address in the application and bank details via email, if applicable).
- [X] I understand that an agreed upon percentage of each milestone will be paid in vested DOT, to the Polkadot address listed in the application.
- [X] I am aware that, in order to receive a grant, I (and the entity I represent) have to successfully complete a KYC/KYB check.
- [X] The software delivered for this grant will be released under an open-source license specified in the application.
- [X] The initial PR contains only one commit (squash and force-push if needed).
- [X] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md)).
- [ ] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@_______:matrix.org` (change the homeserver if you use a different one)

### CODE REVIEW

**Comment by bytemaster333** [comment]
I have read and hereby sign the Contributor License Agreement.

**Comment by bytemaster333** [comment]
recheck

**Review by semuelle — COMMENTED** [review]
Thank you for the application, @bytemaster333. I have two follow-up questions:

1. Are 4,295+ installs "significant interest"? Over what time period are these installs and how does that compare to other vscode plugins?
2. Why do you think the previous plugins were abandoned? Given that there is little money in that market, I'm curious how you see your chances of the community supporting this project long-term.

**Comment by PieWol** [comment]
Hey @bytemaster333 ,
do you have any thoughts on the points @semuelle raised?

**Comment by bytemaster333** [comment]
Hey @semuelle, thanks for the review. Unfortunately, the Visual Studio Marketplace does not show install timelines or activity graphs, which makes it difficult to analyze the time period of installs. However, looking at the available data:

The two extensions have accumulated 4,316 total installs over different periods:

Everstake plugin:
- Active maintenance period: Only 20 days (Sept 5-25, 2019)
- Version history: 5 releases (v0.1.3 through v0.1.7) all within September 2019
- Total installs: 1,610 despite being abandoned after just 20 days of updates

Parity plugin:
- Active maintenance period: 7 months (May-December 2020)
- Version history: 5 releases (v0.4.1 through v0.5.2)
- Total installs: 2,706 despite being abandoned after 7 months

For comparison with other blockchain-focused VSCode extensions:

- Official Cairo language extension by StarkWare Industries: 7,847 installs (https://marketplace.visualstudio.com/items?itemName=starkware.cairo1)
- Official Move language extension by Mysten Labs: 5,715 installs (https://marketplace.visualstudio.com/items?itemName=mysten.move)
- ink! Analyzer: 639 installs (https://marketplace.visualstudio.com/items?itemName=ink-analyzer.ink-analyzer)
- Official Cadence language extension by Flow Blockchain: 7,749 installs (https://marketplace.visualstudio.com/items?itemName=onflow.cadence)

What's notable is that both Substrate extensions achieved their install numbers despite very short maintenance windows (20 days and 7 months respectively) and have continued to accumulate installs even in their deprecated state. This suggests there's a persistent need for Substrate development tooling in VSCode, even if developers are installing outdated extensions due to lack of alternatives. The fact that these abandoned extensions have achieved comparable install numbers to actively maintained official extensions from major chains further supports this observation.

When I first drafted this proposal on November 22, 2024, the combined download count was 4,295. Now, as of December 15, the total has reached 4,316 - meaning 21 "new developers" have installed these deprecated extensions in just 23 days. This averaging to nearly one download per day for extensions that were abandoned in 2019 and 2020 is particularly remarkable, suggesting that Substrate developers are still actively seeking VSCode tooling support, even if it means using outdated extensions. I have waited to reply couple of weeks to observe this.

Now for the second question,

Both extensions were developed during Substrate's early evolution (2019-2020) when the ecosystem was rapidly changing. The frequent breaking changes in Substrate during this period would have made maintenance extremely resource-intensive. Parity's extension was discontinued with a simple note "This repo is not supported anymore. There are no replacement," suggesting that maintaining a developer tool became deprioritized as teams focused on more critical infrastructure development during Substrate's early days.

For long-term sustainability of my extension, I'm planning several technical approaches:

- Building with modern, stable Substrate APIs that are less likely to require frequent major updates
- Implementing a modular architecture where components can be updated independently
- Comprehensive automated testing to reduce maintenance burden

While there will not be direct monetization opportunities, initial development will be funded through this grant. I will actively develop and maintain the project for about 1 year. After 1 year, i will share comprehensive metrics including download counts, developer feedbacks, and feature usage patterns with the community. Based on adoption metrics and community interest, i will be pursuing follow-up proposals for major updates or new feature. I will also fix any criticial bugs or changes after one year but can not guarantee the major releases. I'm confident about the community's demand for this project - there's a clear need shown by developers still installing outdated tools. After completing the initial development, I plan to actively reach out to Substrate developers through various channels (Discord, Element, forums, etc) to ensure the tool addresses their needs. I believe that after one year I will have a well-adopted extension with strong metrics demonstrating its value to the ecosystem. This proven track record will make a compelling case for follow-up proposals to support ongoing development and new features.

File: applications/vscode_substrate_extension.md [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+
+I'm Salih Touner, a Linux Administrator in a private company with a degree in Electrical and Electronics Engineering. I’ve been working on private software projects for the past two months and have a strong interest in blockchain solutions. 
+
+Recently, I've developed a solution for Stellar ecosystem and development of project is about to finish. It helps you to real-time monitoring and visualization system for Soroban CLI commands on the Stellar network, powered by the ELK stack.
+
+- Soroban Command Insights: https://communityfund.stellar.org/project/soroban-command-insights-sci  
+
+### Team Code Repos
+
+- https://github.com/bearstore333
```
Comment (semuelle) [inline applications/vscode_substrate_extension.md:247]: This account does not exist

**Review by semuelle — COMMENTED** [review]
Thanks for the update, @bytemaster333. While it would be nice to have an established VSCode extension, I think it takes a lot of work and resources to develop and maintain the code and to foster a community around it, especially in our fast-paced industry. I don't see you having a lot of experience in these areas, so I won't support the project. But I will share your proposal with the rest of the committee, who might disagree with me.

**Comment by PieWol** [comment]
Hey @bytemaster333 ,
I appreciate the time and effort you put into this application but sadly I have to agree with with @semuelle here. I won't support it either. Still best of luck for your future projects :)

**Comment by keeganquigley** [comment]
Thanks for the discussion @bytemaster333 unfortunately the application wasn't able to garner enough approvals, therefore I will go ahead and close it. While we applaud your efforts, there doesn't seem to be enough interest/support around a VSCode extension at this time. Perhaps this will change in the future. I thank you for your time & effort, and hope you intend to keep building in our ecosystem.

**Pull request closed without merging**